src/Command/Pull/PullCommandBase.php

```diff
@@ -507,14 +507,21 @@ private function promptChooseDatabases(
   }
 
   protected function runComposerScripts(callable $outputCallback = NULL): void {
-    if (file_exists($this->dir . '/composer.json') && $this->localMachineHelper->commandExists('composer')) {
-      $this->checklist->addItem("Installing Composer dependencies");
-      $this->composerInstall($outputCallback);
-      $this->checklist->completePreviousItem();
+    if (!file_exists(Path::join($this->dir, 'composer.json'))) {
+      $this->logger->notice('composer.json file not found. Skipping composer install.');
```

Comment:
I'm not sure that's the behavior we want. The codebase will have `composer.json` and `composer.lock` + the `vendor` dir. That's the issue.

Reply:
See details at https://acquia.atlassian.net/browse/CLI-1181?focusedCommentId=7939845

Reply:
Right, that check for `vendor` comes later. The check for `composer.json` was from before all this and it's something we want to preserve.

Reply:
Removed my comment. That's a public repo 🤦

Reply:
Does my comment make sense? I think the logic is correct. We should not run composer install if composer.json is absent.

Reply:
Yes agreed, the current behavior to not run our logic when `composer.json` is missing is something we want to preserve.

What I was specifically trying to focus on here is the current issue we've discussed with the Trials use case which will have what we need (like a `composer.json` file) and also what we don't expect (a `vendor` dir).

I understand you have covered this here and think we should be good:

```
if (file_exists(Path::join($this->dir, 'vendor'))) {
      $this->io->note('Composer dependencies already installed. Skipping composer install.');
      return;
}
```